### PR TITLE
2.0.x Logging Improvements

### PR DESCRIPTION
**Pull request by RPangrle**
* Move all error logging information into the "message" section
* Add logging for keychain item removal
* Fix some confusing logging around NTLM auth

### CODE REVIEW

**Comment by RPangrle** [comment]
@RandalliLama

File: ADALiOS/ADALiOS/ADKeychainTokenCacheStore.m [hunk 1]
```diff
@@ -536,6 +554,15 @@ -(void) removeAllForUser: (NSString*) userId
                    error: (ADAuthenticationError* __autoreleasing*) error
 {
     API_ENTRY;
+    if (mSharedCache)
+    {
+        AD_LOG_ERROR(@"******** removeAllWithError is being called. This will remove all tokens from the ADAL shared cache that your application has access to. ********", AD_ERROR_UNEXPECTED, nil);
```
Comment (jasoncoolmax) [inline ADALiOS/ADALiOS/ADKeychainTokenCacheStore.m]: Shouldn't the message be "removeAllForUser is being called"? :P

Shall we add the similar logging to method [removeAllForUser: error:] below? :P

Comment (RPangrle) [inline ADALiOS/ADALiOS/ADKeychainTokenCacheStore.m]: It's supposed to be on removeAllWithError: I just copied & pasted into the wrong spot. This removeAllForUser API is a private thing just for the broker. :-/

Comment (jasoncoolmax) [inline ADALiOS/ADALiOS/ADKeychainTokenCacheStore.m]: I see :)

**Comment by jasoncoolmax** [comment]
I signed off :)

**Pull request merged**